recipes/lely-core/all/conanfile.py

```diff
@@ -0,0 +1,242 @@
+        "obj-name": [True, False],
+        "obj-upload": [True, False],
+        "sdev": [True, False],
+        "csdo": [True, False],
+        "rpdo": [True, False],
+        "tpdo": [True, False],
+        "mpdo": [True, False],
+        "sync": [True, False],
+        "time": [True, False],
+        "emcy": [True, False],
+        "lss": [True, False],
+        "wtm": [True, False],
+        "master": [True, False],
+        "ng": [True, False],
+        "nmt-boot": [True, False],
+        "nmt-cfg": [True, False],
+        "gw": [True, False],
+        "gw-txt": [True, False],
+        "coapp-master": [True, False],
+        "coapp-slave": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "rt": True,
+        "threads": True,
+        "ecss-compliance": False,
+        "errno": True,
+        "malloc": True,
+        "stdio": True,
+        "cxx": True,
+        "daemon": True,
+        "diag": True,
+        "canfd": True,
+        "dcf": True,
+        "dcf-restore": True,
+        "obj-default": True,
+        "obj-file": True,
+        "obj-limits": True,
+        "obj-name": True,
+        "obj-upload": True,
+        "sdev": True,
+        "csdo": True,
+        "rpdo": True,
+        "tpdo": True,
+        "mpdo": True,
+        "sync": True,
+        "time": True,
+        "emcy": True,
+        "lss": True,
+        "wtm": True,
+        "master": True,
+        "ng": True,
+        "nmt-boot": True,
+        "nmt-cfg": True,
+        "gw": True,
+        "gw-txt": True,
+        "coapp-master": True,
+        "coapp-slave": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                "This Conan version of this package is only compatible with Linux. "
+                "Windows requires proprietary software from https://www.ixxat.com/technical-support/support/windows-driver-software "
+                "hence support for it will be skipped for now "
+            )
+        if self.settings.compiler != "gcc":
+            raise ConanInvalidConfiguration(
+                "This library can only be compiled with GCC currently"
+            )
+
+    def source(self):
+        get(
+            self,
+            **self.conan_data["sources"][self.version],
+            destination=self.source_folder,
+            strip_root=True,
+        )
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def generate(self):
+        at_toolchain = AutotoolsToolchain(self)
+        at_toolchain.generate()
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def build(self):
+        args = [
+            "--disable-cython",
```

Comment:
Please, move it to AutotoolsToolchain with configure_args: https://docs.conan.io/en/latest/reference/conanfile/tools/gnu/autotoolstoolchain.html#attributes